### DISCUSSION TITLE
o1-preview support？

### DISCUSSION THREAD

**Discussion by tao-shen** [open-webui/open-webui#5388] [Q&A]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by ghost** [comment]
I think it's because o1-preview is not yet returned from this endpoint:

`https://api.openai.com/v1/models`

**Comment by crzroot** [comment]
yes, mine got an error as well
Error message: Oops! There was a problem connecting to the O1-mini.
External: 400, message='Bad Request', url='https://aihubmix.com/v1/chat/completions

**Reply by TianzeWang** [reply]
Likewise

**Comment by m8dhouse** [comment]
API use is currently limited to Tier 5 developer users (was in the openai announcement email)

**Reply by goodtiger** [reply]
I have the Tier 5  account ，but still get the 400 Bad request

**Reply by ghost** [reply]
there's a better discussion over here: https://github.com/open-webui/open-webui/discussions/5390

**Reply by goodtiger** [reply]
> there's a better discussion over here: #5390
Thanks

**Comment by wstlxx** [comment]
Well, I can confirm that o1-preview and o1-mini models are supported already.
the response gives thinking process at first then the remaining answer. 
I use a 3rd party api provider and not sure how it works with openai official api, 
but the non-stream o1 family model response can be handled within openwebui

**Reply by xiao-zy19** [reply]
I guess the third-party APIs are probably faking the streaming process. It's also possible that they are converting ChatGPT web into their o1 model APIs.

**Comment by TweedBeetle** [comment]
I am also tier 5 but also get the following error:

```
Uh-oh! There was an issue connecting to o1-mini.
External: 400, message='Bad Request', url='https://api.openai.com/v1/chat/completions'
```

I can confirm that o1 generally works for me via api

**Comment by ill-yes** [comment]
Any updates on this?

**Comment by petergerten** [comment]
same for me - can't use o1-preview (azure) - getting 400, message='Bad Request'
Other models work without issues. And I can use o1-preview via API no problem.
